src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,208 @@
+    setHistoricalRecords(response.data.results);
+  }
+
+  function getDropZone() {
+    return (
+      <Dropzone
+        onProcessUpload={handleFileUpload}
+      />
+    );
+  }
+
+  function getCollapsibleTitle() {
+    return <span className="font-weight-bold">Processing History</span>;
+  }
+
+  function getAlert() {
+    let variant; let
+      message;
+    if (submitSuccess == null) {
+      return null;
+    }
+
+    if (submitSuccess == true) {
+      variant = 'success';
+      message = 'Successfully submitted task for bulk operation';
+    } else if (submitSuccess == false) {
+      variant = 'danger';
+      message = 'Failed to submit task for processing';
+    }
+
+    return <Alert variant={variant} dismissible onClick={() => { setSubmitSuccess(null); }}>{message}</Alert>;
+  }
+
+  function getTable() {
+    const parsed = Papa.parse(fileContent, {
+      header: true,
+      skipEmptyLines: true,
+    });
+    const columns = parsed.meta.fields.map(fieldName => ({
+      Header: fieldName, accessor: fieldName, width: 400, size: 400,
+    }));
+
+    return (
+      <div data-testid="file-preview" className="data-container-excel">
+        <div className="row justify-content-between border-bottom">
+          <div className="col-auto">
+            <span className="font-weight-bold">{file.name}</span><br />
+            {parsed.data.length} rows - {file.size} B
+          </div>
+          <div className="col-auto">
+            <button data-testid="process-file" className="btn btn-outline-primary mr-2" onClick={handleSubmit}>
+              Process File
+            </button>
+            <button data-testid="upload-new" className="btn btn-outline-primary" onClick={handleUploadNew}>
+              Upload New File
+            </button>
+          </div>
+        </div>
+
+        <div className="mt-3">
+          <DataTable
+            isPaginated
+            initialState={{
+              pageSize: 2,
+            }}
+            itemCount={parsed.data.length}
+            data={parsed.data}
+            columns={columns}
+          >
+            <DataTable.Table />
+            <DataTable.EmptyTable content="No results found" />
+            <DataTable.TableFooter />
+          </DataTable>
+        </div>
+      </div>
+    );
+  }
+
+  useEffect(() => {
+    fetchBulkOpTasks().then(() => setIsLoading(false));
+  }, []);
+
+  if (isLoading) {
+    return (
+      <div className="bulk-operations-spinner">
+        <LoadingSpinner />
+      </div>
+    );
+  }
+
+  return (
+    <div className="mx-5 my-5">
+      {getAlert()}
+      <SelectMenu className="mb-3" defaultMessage="Choose a Bulk Operation">
+        {
+                    Object.entries(availableOperations).map(([slug, title]) => (
+                      <MenuItem actionid={slug} defaultSelected={slug === bulkOperationId} onClick={e => setBulkOperationId(e.currentTarget.getAttribute('actionid'))}>
```

Comment:
If bulkOperationId is slug, please rename it so.

Reply:
renamed `slug` -> `id`